energy_analyse2020.10.16_213101.R

```diff
@@ -0,0 +1,177 @@
+#set path of data and get all the subforders
+#change the path where you put the data
+setwd("D:/Rdata/2020.10.16_213101/data/nexus6p") 
+#change the path where you put the data
+projectpath="D:/Rdata/2020.10.16_213101/data/nexus6p"
+
+library(stringr)
+
+#for app with critical
+criticalfiles<-list.files(pattern = "subjects_critical")
+name_cri_chro <- strsplit(criticalfiles, "-")
+name_cri_chro <- sapply(name_cri_chro, "[", 2)
+appbrowser_cri_chro <- paste(criticalfiles, "chrome", sep = "/")
+androidappbrowser_cri_chro<- paste(appbrowser_cri_chro, "android", sep = "/")
+trepnappbrowser_cri_chro<- paste(appbrowser_cri_chro, "trepn", sep = "/")
+batterystatsappbrowser_cri_chro<- paste(appbrowser_cri_chro, "batterystats", sep = "/")
+androidappbrowser_cri_chro<- paste(appbrowser_cri_chro, "android", sep = "/")
+completetrepath_cri_chro<-paste(projectpath, trepnappbrowser_cri_chro, sep = "/")
+completebatpath_cri_chro<-paste(projectpath, batterystatsappbrowser_cri_chro, sep = "/")
+completeandpath_cri_chro<-paste(projectpath, androidappbrowser_cri_chro, sep = "/")
+timecal_cri_chro=vector()
+memory_usage_cri_chro=vector()
+energy_cri_chro=vector()
+CPU_utilisation_cri_chro=vector()
+'#get time_cri_chro
+for (i in 1:25) {
+  fileswithoutagg_cri_chro<-list.files(completetrepath_cri_chro[i], pattern = "ENU")
+  timesum_cri_chro= 0
+  for(j in 1:10){
+    openpath<-paste(completetrepath_cri_chro[i], fileswithoutagg_cri_chro[j], sep = "/")
+    data<-read.csv(openpath)
+    timesum_cri_chro<-timesum_cri_chro+max(data[,1])
+  }
+  timecal_cri_chro[i]<-timesum_cri_chro/10
+}'
+#get energy
+for (i in 1:25) {
+  filesbattery_cri_chro<-list.files(completebatpath_cri_chro[i], pattern = "Agg")
+  openpath<-paste(completebatpath_cri_chro[i], filesbattery_cri_chro, sep = "/")
+  data<-read.csv(openpath)
+  energy_cri_chro[i]<-data[,1]
+}
+'#get memory_use
+for (i in 1:25) {
+  openpath<-paste(completetrepath_cri_chro[i], "Aggregated.csv", sep = "/")
+  #  print(openpath)
+  data<-read.csv(openpath)
+  memory_usage_cri_chro[i]<-max(data[,1])
+  #  print(memory_usage[i])
+}
+#ger CPU_utilisation
+for (i in 1:25) {
+  openpath<-paste(completeandpath_cri_chro[i], "Aggregated.csv", sep = "/")
+  #print(openpath)
+  data<-read.csv(openpath)
+  CPU_utilisation_cri_chro[i]<-max(data[,1])
+  #  print(CPU_utilisation[i])
+}'
+'trainSet_cri_chro <-data.frame(name_cri_chro,timecal_cri_chro,energy_cri_chro,memory_usage_cri_chro,CPU_utilisation_cri_chro)
+trainSet_cri_chro'
+energy_cri_chro_mean<-mean(energy_cri_chro)
+energy_cri_chro_mean
+energy_cri_chro_median<-median(energy_cri_chro)
+energy_cri_chro_median
+energy_cri_chro_var<-var(energy_cri_chro)
+energy_cri_chro_var
+energy_cri_chro_sd<-sd(energy_cri_chro)
+energy_cri_chro_sd
+energy_cri_chro_range<-range(energy_cri_chro)
+energy_cri_chro_range
+energy_cri_chro_coeffofvariation<-100*sd(energy_cri_chro)/mean(energy_cri_chro)
+energy_cri_chro_coeffofvariation
+
+#for app with noncritical

```

Comment:
This is pretty much identical to the code above. Can this be merged to make it more readable and ensure critical and non critical are treated in the same way?

Reply:
It is a first feeling to feel about the data, not sure it necessary for the report